contracts/MedianOracle.sol

```diff
@@ -35,6 +35,7 @@ contract MedianOracle is Ownable, IOracle {
     event ProviderAdded(address provider);
     event ProviderRemoved(address provider);
     event ReportTimestampOutOfRange(address provider);
+    event ProviderReportPushed(address provider, uint256 payload);
```

Comment:
Can we add the timestamp as well, so that we can really rely on the log instead of getting the timestamp from the block?

Reply:
Done. and added a test